docs/bazel.md

```diff
@@ -0,0 +1,58 @@
+## Bazel in Pytorch/XLA
+
+[Bazel](https://bazel.build/) is a free software tool used for the automation of building and testing software. [TensorFlow](https://www.tensorflow.org/http) and [OpenXLA](https://github.com/openxla/xla) both use it, which makes it a good fit for PyTorch/XLA as well.
+
+## How to build XLA libraries
+
+Building the libraries is simple:
+
+```bash
+bazel build //third_party/xla_client/...
+```
+
+Bazel is configred via `.bazelrc`, but it can also take flags on the command line.
+
+```bash
+bazel build --config=rbe_cpu_linux_py39 //third_party/xla_client/...
+bazel build --config=rbe_linux_cuda11.8_nvcc_py3.9 //third_party/xla_client/...
+```
+
+The `rbe` configurations build remotely and are cahced and usually faster, but require
+authentication with gcloud. See `.bazelrc` for other `rbe` options.
+
+## Running tests
+
+Bazel is a test plafrom too, making it easy to run tests:
+
+```bash
+bazel test //third_party/xla_client/...
+```
+
+Tests can also be executed remotely in the same fashion as builds.
+
+## Code coverage
+
+When running tests, it can be useful to calculate code coverage.
+
+```bash
+bazel coverage //third_party/xla_client/...
```

Comment:
I used to use blaze. How much difference are there between bazel and blaze besides the fact that bazel is the open source version of blaze?

Reply:
THere should be no differences. Bazel has some quirks (not monorepo) so that's a bit different, and some coverage html support is not there, but when it works it should be equivalent.